vm/src/builtins/union.rs

```diff
@@ -156,6 +167,8 @@ fn flatten_args(args: PyTupleRef, vm: &VirtualMachine) -> PyTupleRef {
     for arg in &args {
         if let Some(pyref) = arg.downcast_ref::<PyUnion>() {
             flattened_args.extend(pyref.args.iter().cloned());
+        } else if arg.payload_if_subclass::<PyNone>(vm).is_some() {
```

Comment:
```suggestion
        } else if vm.is_none(arg) {
```
`None` is singleton in python. we check its id instead of its type.